Resource crash `junos_security_nat_source_pool` when `port_range` is used

Howdy, I'm looking at this provider to move some custom configuration tool, so I have a bunch of config to migrate.  Here is what I'm up to first.

```
resource "junos_security_nat_source_pool" "wireguard_v4" {
  name       = "wireguard_v4"
  address    = ["xxx.xxx.xxx.xxx/32"]
  port_range = "11111"
}
```

With this resource, I get the following crash.
```
Stack trace from the terraform-provider-junos_v1.26.0 plugin:

panic: runtime error: index out of range [1] with length 1

goroutine 60 [running]:
github.com/jeremmfr/terraform-provider-junos/junos.validateSourcePoolPortRange.func1({0xfade60?, 0xc0001ce130?}, {0xc0005a7fc0, 0x1, 0x1})
	/home/runner/work/terraform-provider-junos/terraform-provider-junos/junos/resource_security_nat_source_pool.go:474 +0xc45
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Schema).validateFunc(0xc00089f9da?, {0xfade60?, 0xc0001ce130?}, {0x10b7830?, 0x10?}, {0xc0005a7fc0?, 0x1, 0x1})
	/home/runner/go/pkg/mod/github.com/jeremmfr/terraform-plugin-sdk/v2@v2.13.1-0.20220411071827-346044975466/helper/schema/schema.go:566 +0xfa
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.schemaMap.validatePrimitive(0xc0008bef68?, {0x10b7830, 0xa}, {0xfade60, 0xc0005a79d0}, 0xc00051b540, 0xc000572bd0?, {0xc0005a7fc0, 0x1, 0x1})
	/home/runner/go/pkg/mod/github.com/jeremmfr/terraform-plugin-sdk/v2@v2.13.1-0.20220411071827-346044975466/helper/schema/schema.go:2294 +0xa08
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.schemaMap.validateType(0xc0005a7d70?, {0x10b7830?, 0x60e72a?}, {0xfade60?, 0xc0005a79d0?}, 0xc00051b540, 0xfd1360?, {0xc0005a7fc0, 0x1, 0x1})
	/home/runner/go/pkg/mod/github.com/jeremmfr/terraform-plugin-sdk/v2@v2.13.1-0.20220411071827-346044975466/helper/schema/schema.go:2321 +0x14a
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.schemaMap.validate(0xff6440?, {0x10b7830, 0xa}, 0xc00051b540, 0x11c?, {0xc0005a7fc0, 0x1, 0x1})
	/home/runner/go/pkg/mod/github.com/jeremmfr/terraform-plugin-sdk/v2@v2.13.1-0.20220411071827-346044975466/helper/schema/schema.go:1731 +0x6ee
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.schemaMap.validateObject(0x11?, {0x0, 0x0}, 0x12?, 0x0?, {0x1838228?, 0x0, 0x0})
	/home/runner/go/pkg/mod/github.com/jeremmfr/terraform-plugin-sdk/v2@v2.13.1-0.20220411071827-346044975466/helper/schema/schema.go:2167 +0x591
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.schemaMap.Validate(...)
	/home/runner/go/pkg/mod/github.com/jeremmfr/terraform-plugin-sdk/v2@v2.13.1-0.20220411071827-346044975466/helper/schema/schema.go:786
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Validate(0xc00051e000, 0xc0003629f0?)
	/home/runner/go/pkg/mod/github.com/jeremmfr/terraform-plugin-sdk/v2@v2.13.1-0.20220411071827-346044975466/helper/schema/resource.go:916 +0x4d
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Provider).ValidateResource(0x12664d8?, {0xc000604680, 0x1e}, 0xd?)
	/home/runner/go/pkg/mod/github.com/jeremmfr/terraform-plugin-sdk/v2@v2.13.1-0.20220411071827-346044975466/helper/schema/provider.go:255 +0xf2
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ValidateResourceTypeConfig(0xc00011c0a8, {0x1266430?, 0xc0002569c0?}, 0xc000018618)
	/home/runner/go/pkg/mod/github.com/jeremmfr/terraform-plugin-sdk/v2@v2.13.1-0.20220411071827-346044975466/helper/schema/grpc_provider.go:246 +0x1a9
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ValidateResourceTypeConfig(0xc0004bb360, {0x12664d8?, 0xc0005720c0?}, 0xc000256740)
	/home/runner/go/pkg/mod/github.com/hashicorp/terraform-plugin-go@v0.8.0/tfprotov5/tf5server/server.go:687 +0x2f5
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ValidateResourceTypeConfig_Handler({0x1088aa0?, 0xc0004bb360}, {0x12664d8, 0xc0005720c0}, 0xc000399380, 0x0)
	/home/runner/go/pkg/mod/github.com/hashicorp/terraform-plugin-go@v0.8.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:277 +0x170
google.golang.org/grpc.(*Server).processUnaryRPC(0xc000142700, {0x1269058, 0xc0006829c0}, 0xc00025ea20, 0xc000117680, 0x17fa2f0, 0x0)
	/home/runner/go/pkg/mod/google.golang.org/grpc@v1.45.0/server.go:1282 +0xccf
google.golang.org/grpc.(*Server).handleStream(0xc000142700, {0x1269058, 0xc0006829c0}, 0xc00025ea20, 0x0)
	/home/runner/go/pkg/mod/google.golang.org/grpc@v1.45.0/server.go:1619 +0xa1b
google.golang.org/grpc.(*Server).serveStreams.func1.2()
	/home/runner/go/pkg/mod/google.golang.org/grpc@v1.45.0/server.go:921 +0x98
created by google.golang.org/grpc.(*Server).serveStreams.func1
	/home/runner/go/pkg/mod/google.golang.org/grpc@v1.45.0/server.go:919 +0x28a

Error: The terraform-provider-junos_v1.26.0 plugin crashed!
```

Looking at the srx config I have `set security nat source pool wireguard_v4 port range 11111`.  Please advise.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -5,6 +5,8 @@ ENHANCEMENTS:
 
 BUG FIXES:
 
+* resource/`junos_security_nat_source_pool`: fix crash and set correctly with a single port in `port_range`, add format precision in docs (Fixes #382)
+
 ## 1.26.0 (April 13, 2022)
 
 ENHANCEMENTS:

--- a/docs/resources/security_nat_source_pool.md
+++ b/docs/resources/security_nat_source_pool.md
@@ -40,7 +40,8 @@ The following arguments are supported:
 - **port_overloading_factor** (Optional, Number)  
   Port overloading factor for each IP.
 - **port_range** (Optional, String)  
-  Range of port for source nat.
+  Range of port for source nat.  
+  Format need to match `\d+(-\d+)?` with minimum low port 1024 and maximum high port 65535.
 - **routing_instance** (Optional, String)  
   Name of routing instance for switch with nat.
 

--- a/junos/resource_security_nat_source_pool.go
+++ b/junos/resource_security_nat_source_pool.go
@@ -3,6 +3,7 @@ package junos
 import (
 	"context"
 	"fmt"
+	"regexp"
 	"strconv"
 	"strings"
 
@@ -338,7 +339,11 @@ func setSecurityNatSourcePool(d *schema.ResourceData, m interface{}, jnprSess *N
 	}
 	if d.Get("port_range").(string) != "" {
 		rangePort := strings.Split(d.Get("port_range").(string), "-")
-		configSet = append(configSet, setPrefix+" port range "+rangePort[0]+" to "+rangePort[1])
+		if len(rangePort) > 1 {
+			configSet = append(configSet, setPrefix+" port range "+rangePort[0]+" to "+rangePort[1])
+		} else {
+			configSet = append(configSet, setPrefix+" port range "+rangePort[0])
+		}
 	}
 	if d.Get("routing_instance").(string) != "" {
 		configSet = append(configSet, setPrefix+" routing-instance "+d.Get("routing_instance").(string))
@@ -455,14 +460,16 @@ func validateSourcePoolPortRange() schema.SchemaValidateDiagFunc {
 	return func(i interface{}, path cty.Path) diag.Diagnostics {
 		var diags diag.Diagnostics
 		v := i.(string)
-		vSplit := strings.Split(v, "-")
-		if len(vSplit) < 2 {
+		if ok := regexp.MustCompile(`^\d+(-\d+)?$`).MatchString(v); !ok {
 			diags = append(diags, diag.Diagnostic{
 				Severity:      diag.Error,
-				Summary:       fmt.Sprintf("missing range separtor - in %s", v),
+				Summary:       fmt.Sprintf(`expected value of port_range to match regular expression \d+(-\d+)?, got %v`, v),
 				AttributePath: path,
 			})
+
+			return diags
 		}
+		vSplit := strings.Split(v, "-")
 		low, err := strconv.Atoi(vSplit[0])
 		if err != nil {
 			diags = append(diags, diag.Diagnostic{
@@ -470,14 +477,22 @@ func validateSourcePoolPortRange() schema.SchemaValidateDiagFunc {
 				Summary:       err.Error(),
 				AttributePath: path,
 			})
+
+			return diags
 		}
-		high, err := strconv.Atoi(vSplit[1])
-		if err != nil {
-			diags = append(diags, diag.Diagnostic{
-				Severity:      diag.Error,
-				Summary:       err.Error(),
-				AttributePath: path,
-			})
+		high := low
+		if len(vSplit) > 1 {
+			var err error
+			high, err = strconv.Atoi(vSplit[1])
+			if err != nil {
+				diags = append(diags, diag.Diagnostic{
+					Severity:      diag.Error,
+					Summary:       err.Error(),
+					AttributePath: path,
+				})
+
+				return diags
+			}
 		}
 		if low > high {
 			diags = append(diags, diag.Diagnostic{
@@ -493,10 +508,10 @@ func validateSourcePoolPortRange() schema.SchemaValidateDiagFunc {
 				AttributePath: path,
 			})
 		}
-		if high > 63487 {
+		if high > 65535 {
 			diags = append(diags, diag.Diagnostic{
 				Severity:      diag.Error,
-				Summary:       fmt.Sprintf("high(%d) in %s is too big (max 63487)", high, v),
+				Summary:       fmt.Sprintf("high(%d) in %s is too big (max 65535)", high, v),
 				AttributePath: path,
 			})
 		}

--- a/junos/resource_security_nat_source_test.go
+++ b/junos/resource_security_nat_source_test.go
@@ -135,6 +135,13 @@ resource "junos_security_nat_source_pool" "testacc_securitySNATPool" {
   pool_utilization_alarm_raise_threshold = 80
   pool_utilization_alarm_clear_threshold = 60
 }
+resource "junos_security_nat_source_pool" "testacc_securitySNATPool2" {
+  name             = "testacc_securitySNATPool2"
+  description      = "testacc securitySNATPool2"
+  address          = ["192.0.2.2/32"]
+  routing_instance = junos_routing_instance.testacc_securitySNAT.name
+  port_range       = "1300"
+}
 
 resource "junos_security_zone" "testacc_securitySNAT" {
   name = "testacc_securitySNAT"
@@ -205,6 +212,13 @@ resource "junos_security_nat_source_pool" "testacc_securitySNATPool" {
   routing_instance        = junos_routing_instance.testacc_securitySNAT.name
   address_pooling         = "no-paired"
   port_overloading_factor = 3
+}
+resource "junos_security_nat_source_pool" "testacc_securitySNATPool2" {
+  name             = "testacc_securitySNATPool2"
+  description      = "testacc securitySNATPool2"
+  address          = ["192.0.2.2/32"]
+  routing_instance = junos_routing_instance.testacc_securitySNAT.name
+  port_range       = "1300-62000"
 }
 
 resource "junos_security_zone" "testacc_securitySNAT" {

